mlflow/server/handlers.py

```diff
@@ -1835,7 +1835,7 @@ def _validate_non_local_source_contains_relative_paths(source: str):
     while (unquoted := urllib.parse.unquote_plus(source)) != source:
         source = unquoted
     source_path = re.sub(r"/+", "/", urllib.parse.urlparse(source).path.rstrip("/"))
-    if "\x00" in source_path:
+    if "\x00" in source_path or ".." in source_path or ".." in source:
```

Comment:
Can we use `any(".." == part for part in source_path.split("/"))` instead of checking for ".." in `source_path` to improve readability and ensure we are only looking at each segment of the path?

Reply:
```suggestion
    if "\x00" in source_path or any(p == ".." for p in source_path.split("/")):
```

Reply:
@harupy It looks like this still only checks source.path?